Shopify/react-native-performance
# feat(expo-config-plugin): create config plugin for react-native-performance

**Motivation**

Expo is becoming an increasingly popular way of building React-Native projects. It is important that we support their ecosystem in order to keep this library relevant in the RN Community.

**NOTE:** Reverted the [import change](https://github.com/Shopify/react-native-performance/pull/118/files) found here because it seemed to be breaking builds in Vanilla RN and Expo Projects. Please double check this fix works on your end @ElviraBurchik. 

## Description

Fixes N/A

This Pull request adds a plugin folder to the library that has the code for the Expo Config-Plugin. No other code was touched. The purpose of the Config plugin is just to update the code on the library consumer's end

### Test plan

1. Checkout this branch
2. At the root of the project run `yarn` to install all dependancies
3. Run `yarn build` to build the config plugin locally
4. `cd` into the `ExpoFixture` and run `yarn` again to install it's dependancies
5. Run `npx expo prebuild` to do all native code setup
6. After this run the app on both platforms making sure that you can see the logs in Flipper

For iOS:

```bash
npx expo run:ios
```

For Android:

```bash
npx expo run:android
```

## Checklist


- [ ] I have added a decision record entry, PR includes changes to monorepo setup that may require explanation
